pontoon/sync/tasks.py

```diff
@@ -76,12 +84,15 @@ def sync_project(self, project_pk, sync_log_pk, locale=None, no_pull=False, no_c
                 now,
                 locale=locale,
                 no_pull=no_pull,
-                no_commit=no_commit
+                no_commit=no_commit,
+                full_scan=full_scan,
+                vcs_hanged_files=db_project.changed_vcs_locale_files.get(repo.pk, {})
             )
 
     else:
         try:
-            obsolete_vcs_entities, obsolete_vcs_resources = perform_sync_project(db_project, now)
+            obsolete_vcs_entities, obsolete_vcs_resources = perform_sync_project(db_project, now,
+                                                                                 db_project.changed_vcs_source_files)
```

Comment:
We pass the entire `db_project`, no need to pass `db_project.changed_vcs_source_files` separately. Unless we move `changed_source_files` to `vcs_project`.